lms/models/h_group.py

```diff
@@ -1,59 +1,36 @@
-from typing import NamedTuple
+import sqlalchemy as sa
 
-from lms.models._hashed_id import hashed_id
+from lms.db import BASE, TimestampedModelMixin
 
 MAX_GROUP_NAME_LENGTH = 25
 
 
-class HGroup(NamedTuple):
-    name: str
-    authority_provided_id: str
-    type: str
+class HGroup(TimestampedModelMixin, BASE):
+    __tablename__ = "h_group"
+    __table_args__ = (sa.UniqueConstraint("authority_provided_id"),)
```

Comment:
```suggestion
```

I think you can just add `unique=True` to the `authority_provided_id = sa.Column(...)` below

Reply:
Yep initially needed to name the index to use the `on_conflict` but after updating to use the ORM that's not needed